src/core/raster/qgsmultibandcolorrenderer.cpp

```diff
@@ -416,3 +417,121 @@ QList<int> QgsMultiBandColorRenderer::usesBands() const
+  //  isDefaultCombination = isDefaultCombination && (
+  //      mRedContrastEnhancement->contrastEnhancementAlgorithm() == QgsContrastEnhancement::StretchToMinimumMaximum &&
+  //      mGreenContrastEnhancement->contrastEnhancementAlgorithm() == QgsContrastEnhancement::StretchToMinimumMaximum &&
+  //      mBlueContrastEnhancement->contrastEnhancementAlgorithm() == QgsContrastEnhancement::StretchToMinimumMaximum
+  //    );
+  //  // compute raster statistics (slow) only if true the previous conditions
+  //  if ( isDefaultCombination)
+  //  {
+  //    QgsRasterBandStats statRed = bandStatistics(1, QgsRasterBandStats::Min|QgsRasterBandStats::Max);
+  //    isDefaultCombination = isDefaultCombination && (
+  //        ( mRedContrastEnhancement->minimumValue() == statRed.minimumValue &&
+  //          mRedContrastEnhancement->maximumValue() == statRed.maximumValue )
+  //      );
+  //  }
+  //  if ( isDefaultCombination)
+  //  {
+  //    QgsRasterBandStats statGreen = bandStatistics(2, QgsRasterBandStats::Min|QgsRasterBandStats::Max);
+  //    isDefaultCombination = isDefaultCombination && (
+  //          ( mGreenContrastEnhancement->minimumValue() == statGreen.minimumValue &&
+  //            mGreenContrastEnhancement->maximumValue() == statGreen.maximumValue )
+  //      );
+  //  }
+  //  if ( isDefaultCombination)
+  //  {
+  //    QgsRasterBandStats statBlue = bandStatistics(3, QgsRasterBandStats::Min|QgsRasterBandStats::Max);
+  //    isDefaultCombination = isDefaultCombination && (
+  //          ( mBlueContrastEnhancement->minimumValue() == statBlue.minimumValue &&
+  //            mBlueContrastEnhancement->maximumValue() == statBlue.maximumValue )
+  //      );
+  //  }
+  //  if ( isDefaultCombination ):
+  //    return
+
+  // look for RasterSymbolizer tag
+  QDomNodeList elements = element.elementsByTagName( QStringLiteral( "sld:RasterSymbolizer" ) );
+  if ( elements.size() == 0 )
+    return;
+
+  // there SHOULD be only one
+  QDomElement rasterSymbolizerElem = elements.at(0).toElement();
+
+  // add Channel Selection tags
+  // Need to insert channelSelection in the correct sequence as in SLD standard e.g.
+  // after opacity or geometry or as first element after sld:RasterSymbolizer
+  QDomElement channelSelectionElem = doc.createElement( QStringLiteral( "sld:ChannelSelection" ) );
+  elements = rasterSymbolizerElem.elementsByTagName( QStringLiteral( "sld:Opacity" ) );
+  if ( elements.size() != 0 )
+  {
+    rasterSymbolizerElem.insertAfter( channelSelectionElem, elements.at(0) );
+  }
+  else
+  {
+    elements = rasterSymbolizerElem.elementsByTagName( QStringLiteral( "sld:Geometry" ) );
+    if ( elements.size() != 0 )
+    {
+      rasterSymbolizerElem.insertAfter( channelSelectionElem, elements.at(0) );
+    }
+    else
+    {
+      rasterSymbolizerElem.insertBefore( channelSelectionElem, rasterSymbolizerElem.firstChild() );
+    }
+  }
+
+  // for each mapped band
+  QStringList tags;
```

Comment:
this could be a static QStringList, with an initializer list instead of rebuilding with every call

Reply:
ok